### PR TITLE
Add `lxd` group creation to installation process

### PR DESCRIPTION
**Pull request by hamistao**
@ru-fu added `lxd` group creation information as discussed.

### CODE REVIEW

File: doc/installing.md [hunk 1]
```diff
@@ -289,10 +302,19 @@ Access control for LXD is based on group membership.
 The root user and all members of the `lxd` group can interact with the local daemon.
 See {ref}`security-daemon-access` for more information.
 
-If the `lxd` group is missing on your system, create it and restart the LXD daemon.
-You can then add trusted users to the group.
-Anyone added to this group will have full control over LXD.
+On Ubuntu images, the `lxd` group is already created and the main user is automatically added to it. The group is also created during installation if you {ref}`installed LXD from the snap<installing-from-package>`. If the `lxd` group is missing on your system(as might be the case if you {ref}`installed LXD from the sources <installing-from-source>`), create it and restart the LXD daemon:
```
Comment (ru-fu) [inline doc/installing.md:305]:
```suggestion
On Ubuntu images, the `lxd` group already exists and the main user is automatically added to it. The group is also created during installation if you {ref}`installed LXD from the snap<installing-from-package>`. If the `lxd` group is missing on your system (as might be the case if you {ref}`installed LXD from the sources <installing-from-source>`), create it and restart the LXD daemon:
```

File: doc/installing.md [hunk 2]
```diff
@@ -289,10 +302,19 @@ Access control for LXD is based on group membership.
...
 
+```bash
+grep -q "^lxd:" /etc/group && sudo groupadd lxd
+```
+
+No users are added to the group on installation. You can then add trusted users to the group.
```
Comment (ru-fu) [inline doc/installing.md:311]:
```suggestion
No users are added to the group on installation. You must add trusted users to the group so they can use LXD:
```

**Review by ru-fu — COMMENTED** [review]
Thanks, this looks good!
Some small suggestions.

**Review by ru-fu — DISMISSED** [review]
Thank you! :rocket:

File: doc/installing.md [hunk 3]
```diff
@@ -271,6 +278,13 @@ You'll need sub{u,g}ids for root, so that LXD can create the unprivileged contai
 echo "root:1000000:1000000000" | sudo tee -a /etc/subuid /etc/subgid
 ```
 
+By default, only users added to the `lxd` group can interact with the LXD daemon. Installing from source doesn't guarantee that the `lxd` group exists in the system. If you want the current user (or any other user) to be able to interact with the LXD daemon, add it to the `lxd` group:
+
+```bash
+grep -q "^lxd:" /etc/group && sudo groupadd lxd # create the group if needed
+grep -q "^lxd:" /etc/group | grep -q $USER && sudo usermod -aG lxd $USER
```
Comment (simondeziel) [inline doc/installing.md:285]:
```suggestion
getent group lxd >/dev/null || sudo groupadd --system lxd # create the group if needed
getent group lxd | grep -qwF "$USER" || sudo usermod -aG lxd "$USER"
```

File: doc/installing.md [hunk 4]
```diff
@@ -289,10 +303,19 @@ Access control for LXD is based on group membership.
 The root user and all members of the `lxd` group can interact with the local daemon.
 See {ref}`security-daemon-access` for more information.
 
-If the `lxd` group is missing on your system, create it and restart the LXD daemon.
-You can then add trusted users to the group.
-Anyone added to this group will have full control over LXD.
+On Ubuntu images, the `lxd` group already exists and the main user is automatically added to it. The group is also created during installation if you {ref}`installed LXD from the snap<installing-from-package>`. If the `lxd` group is missing on your system (as might be the case if you {ref}`installed LXD from the sources <installing-from-source>`), create it and restart the LXD daemon:
 
+```bash
+grep -q "^lxd:" /etc/group && sudo groupadd lxd
```
Comment (simondeziel) [inline doc/installing.md:309]:
```suggestion
getent group lxd >/dev/null || sudo groupadd --system lxd
```

File: doc/installing.md [hunk 5]
```diff
@@ -289,10 +303,19 @@ Access control for LXD is based on group membership.
...
+```
+
+No users are added to the group on installation. You must add trusted users to the group so they can use LXD:
+
+```bash
+grep -q "^lxd:" /etc/group | grep -q $USER && sudo usermod -aG lxd $USER # adding current user as an example
+```
+
+Anyone added to this group will have full control over LXD.
```
Comment (simondeziel) [inline doc/installing.md:318]: I think we should probably link to [doc/explanation/security.md](https://documentation.ubuntu.com/lxd/en/latest/explanation/security/#local-access-to-the-lxd-daemon) as being a member of that group is akin to being root.

**Review by simondeziel — CHANGES_REQUESTED** [review]

**Comment by tomponline** [comment]
@hamistao are you able to make the changes suggested by @simondeziel ?

**Comment by tomponline** [comment]
@ru-fu should we also include the group instructions in https://documentation.ubuntu.com/lxd/en/latest/tutorial/first_steps/ as we've had a new user follow this who was presumably not on Ubuntu and so got a bit stuck, see https://discourse.ubuntu.com/t/lxd-5-21-1-lts-has-been-released/43823/79

**Comment by ru-fu** [comment]
> @ru-fu should we also include the group instructions in https://documentation.ubuntu.com/lxd/en/latest/tutorial/first_steps/ as 

Yes, that makes sense! Should be a 4th step in the install section - can you add that @hamistao ?

> we've had a new user follow this who was presumably not on Ubuntu and so got a bit stuck, see https://discourse.ubuntu.com/t/lxd-5-21-1-lts-has-been-released/43823/79

Unrelated comment - for `lxd init`, we're saying this can be run without `sudo` for simple configurations:
https://documentation.ubuntu.com/lxd/en/latest/howto/initialize/
Has that changed?

**Comment by tomponline** [comment]
> Unrelated comment - for `lxd init`, we're saying this can be run without `sudo` for simple configurations:
> https://documentation.ubuntu.com/lxd/en/latest/howto/initialize/
> Has that changed?

Its probably still fine as long as user is in lxd group.

**Comment by hamistao** [comment]
@ru-fu @simondeziel @tomponline I have made all requested changes. In the [first steps section](https://documentation.ubuntu.com/lxd/en/latest/tutorial/first_steps/) I only added a command to add a user to the `lxd` group since the group itself is created during snap installation (i.e. step 2 in the stallation process).

**Review by tomponline — APPROVED** [review]

**Review by ru-fu — COMMENTED** [review]
Thank you! :rocket:

**Pull request merged**